### PR TITLE
fix #133 XSS injection (with Htmlpurifier)

### LINKED ISSUE #133: XSS injection in comments, post content and summary
It is possible to exploit XSS injection in comments, post content and summary. All you need is to post this content into forms:

```
<script>alert(123);</script>
```

### PR DESCRIPTION
**Pull request by e-moe**
Q |	A
---------|---------
Bug fix? |	yes
New feature? |	no
BC breaks?	| no
Deprecations? |	no
Tests pass? |	yes
Fixed tickets |	#133 
License |	MIT
Doc | PR

### CODE REVIEW

File: src/AppBundle/Twig/AppExtension.php [hunk 1]
```diff
@@ -28,9 +29,12 @@ class AppExtension extends \Twig_Extension
 {
     private $parser;
 
-    public function __construct(Markdown $parser)
+    private $purifier;
+
+    public function __construct(Markdown $parser, HTMLPurifierExtension $purifier)
```
Comment (stof) [inline src/AppBundle/Twig/AppExtension.php]: injecting a Twig extension into another one does not make sense. you should inject HtmlPurifier itself

Comment (stof) [inline src/AppBundle/Twig/AppExtension.php]: and I think this should be done in the Markdown class itself, by making ``AppBundle\Utils\Markdown`` a safe markdown parser

Comment (e-moe) [inline src/AppBundle/Twig/AppExtension.php]: ok, thanks for comment. I'll try to update this pull request

**Comment by e-moe** [comment]
@stof , thanks for fast response. your comments applied ;)

File: composer.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
         "doctrine/doctrine-fixtures-bundle"    : "~2.2",
         "doctrine/orm"                         : "~2.4",
         "erusev/parsedown"                     : "~1.5",
+        "ezyang/htmlpurifier"                  : "~4.0",
```
Comment (stof) [inline composer.json]: please make it ``~4.7`` instead. We don't want older versions (4.6 makes messy things with permissions in its cache folder)

Comment (e-moe) [inline composer.json]: done

**Comment by e-moe** [comment]
@stof , sorry for disturbing. pull request is updated due to all your notes :)

File: src/AppBundle/Utils/Markdown.php [hunk 3]
```diff
@@ -23,15 +23,19 @@ class Markdown
 {
     private $parser;
 
+    private $purifier;
+
     public function __construct()
     {
         $this->parser = new \Parsedown();
+        $this->purifier = new \HTMLPurifier();
```
Comment (bocharsky-bw) [inline src/AppBundle/Utils/Markdown.php]: Could you move `HTMLPurifier` creation out to a separate class and define it as a service? I think we could create `AppBundle\Utils\Purifier` class for it and then we could inject it into the `Markdown` service. Also, in this case, we could easily use `Purifier` as a standalone service in any part of our demo application.

Comment (javiereguiluz) [inline src/AppBundle/Utils/Markdown.php]: Before doing that, are we sure we'll use the Purifier service anywhere else in the application?

Comment (bocharsky-bw) [inline src/AppBundle/Utils/Markdown.php]: Anyway, I thought it'll be clearer to get separate service for purifier.

Do you think it'll be a bit overhead for available services?

Comment (javiereguiluz) [inline src/AppBundle/Utils/Markdown.php]: I agree with you if we needed to purify other contents in the application. But for now, it doesn't look like we need it. So I prefer to not create that service.

Comment (bocharsky-bw) [inline src/AppBundle/Utils/Markdown.php]: OK, then we could refactor and create it when it's needed.

**Comment by javiereguiluz** [comment]
Thanks for working on this feature @e-moe

**Comment by stof** [comment]
Given you don't use the bundle anymore, you should configure HtmlPurifier to write its cache inside the Symfony cache (where we are sure we can write)

**Comment by e-moe** [comment]
ok, I'll try to create one more pull request to fix HtmlPurifier  cache

**Pull request merged**